src/retcodes.jl

```diff
@@ -0,0 +1,53 @@
+Enumx.@enumx(ReturnCode,Default,Success,Terminated,MaxIters,DtLessThanMin,Unstable,
+      InitialFailure,ConvergenceFailure,Failure)
+
+function Base.Symbol(retcode::ReturnCode)
```

Comment:
This method is already supported bot enums, no?

Reply:
It is. This and the hacky `convert` are what I were playing around with for backwards compatibility since NonlinearSolve defined some of its own enums, but I don't think this is the right solution. Instead, expanding the meaning of success and telling people to use `successful_retcode(sol.retcode)` for things like `Success` or `Terminated` (which is, solve successfully reached steady state and terminated before final time point), `EXACT_SOLUTION_LEFT` (bisection grabbed the left), `FLOATING_POINT_LIMIT` (bisection successfully went to the floating point limit). Trying to slam everything to `Success` loses information, "is any form of success" seems like a better query for things like optimization loops. 

However, I'll loop back to that after I figure out what's going on with downstream tests. DelayDiffEq.jl is being puzzling.